asking about pkg-config problem

I `go get` in window. It's come up with

```
$ go get github.com/sbinet/go-python
# pkg-config --cflags python-2.7
Package python-2.7 was not found in the pkg-config search path.
Perhaps you should add the directory containing `python-2.7.pc'
to the PKG_CONFIG_PATH environment variable
No package 'python-2.7' found
pkg-config: exit status 1
```

This not happen for me in OSX
Now I cant even find `.pc` files in any directories. So I don't know what to add in PKG_CONFIG_PATH

Sorry for asking stupid question
I'm pretty new in window, python, pkg-config

Thank you for very fast response, I've got this (in cygwin)

```
# github.com/sbinet/go-python
In file included from C:/Python27/include/Python.h:58:0,
                 from go-python.h:4,
                 from ..\github.com\sbinet\go-python\capi.go:3:
C:/Python27/include/pyport.h:886:2: error: #error "LONG_BIT definition appears w              rong for platform (bad gcc/glibc config?)."
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?).              "
  ^
```

I have same issue in windows:

```
{user}\AppData\Local\Temp\go-build142102602\github.com\sbinet\go-python\_obj\exceptions.cgo2.o: In function `_cgo_75d15165160c_Cfunc_PySignal_SetWakeupFd':
..\../cgo-gcc-prolog:493: undefined reference to `PySignal_SetWakeupFd'
{user}\AppData\Local\Temp\go-build142102602\github.com\sbinet\go-python\_obj\go-python.o: In function `_gopy_InitModule':

{GOPATH}/github.com/sbinet/go-python/go-python.c:147: undefined reference to `__imp_Py_InitModule4'

{user}\AppData\Local\Temp\go-build142102602\github.com\sbinet\go-python\_obj\go-python.o: In function `_gopy_InitModule3':

{GOPATH}/github.com/sbinet/go-python/go-python.c:152: undefined reference to `__imp_Py_InitModule4'

collect2.exe: error: ld returned 1 exit status
```

This is my config for `cgoflags_windows.go`:

```go
package python

// #cgo CFLAGS: -I{USER}/Anaconda2/include
// #cgo LDFLAGS: -L{USER}/Anaconda2/libs -lpython27
// #include "go-python.h"
import "C"
```

My environments:
`Python 2.7.14 |Anaconda, Inc.| (default, Oct 15 2017, 03:34:40) [MSC v.1500 64 bit (AMD64)] on win32`

@sbinet , any suggest?

I got this problem too,
```
In file included from C:/Python27/include/Python.h:61:0,
                 from ./go-python.h:4,
                 from C:\gopath\src\github.com\sbinet\go-python\capi.go:3:
C:/Python27/include/pyport.h:895:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
```
Have you solved it?

--- a/README.md
+++ b/README.md
@@ -2,6 +2,7 @@ go-python
 =========
 
 [![Build Status](https://drone.io/github.com/sbinet/go-python/status.png)](https://drone.io/github.com/sbinet/go-python/latest)
+[![Build status](https://ci.appveyor.com/api/projects/status/n0ujg8no487a89vo/branch/master?svg=true)](https://ci.appveyor.com/project/sbinet/go-python/branch/master)
 
 Naive `go` bindings towards the C-API of CPython-2.
 

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 go-python
 =========
 
-[![Build Status](https://drone.io/github.com/sbinet/go-python/status.png)](https://drone.io/github.com/sbinet/go-python/latest)
+[![Build Status](https://travis-ci.org/sbinet/go-python.svg?branch=master)](https://travis-ci.org/sbinet/go-python)
 [![Build status](https://ci.appveyor.com/api/projects/status/n0ujg8no487a89vo/branch/master?svg=true)](https://ci.appveyor.com/project/sbinet/go-python/branch/master)
 
 Naive `go` bindings towards the C-API of CPython-2.

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,29 @@
+build: off
+
+clone_folder: c:\gopath\src\github.com\sbinet\go-python
+
+cache:
+  - '%LocalAppData%\\go-build'
+  - '%LocalAppData%\\pip'
+
+branches:
+  only:
+    - master
+
+environment:
+  GOPATH: c:\gopath
+  GOPY_APPVEYOR_CI: '1'
+  GODEBUG: 'cgocheck=0'
+  GOTRACEBACK: 'crash'
+  PYTHONUNBUFFERED: "1"
+  CPYTHON2DIR: "C:\\Python27-x64"
+  PATH: '%GOPATH%\bin;%CPYTHON2DIR%;%CPYTHON2DIR%\\Scripts;C:\msys64\mingw64\bin;C:\msys64\usr\bin\;%PATH%'
+
+stack: go 1.11
+
+build_script:
+  - "%CPYTHON2DIR%\\python --version"
+  - go get -v -t ./...
+
+test_script:
+  - go test ./...

--- a/cgoflags_unix.go
+++ b/cgoflags_unix.go
@@ -1,7 +1,7 @@
+// +build !windows
+
 package python
 
 // #cgo pkg-config: python-2.7
 // #include "go-python.h"
 import "C"
-
-// EOF

--- a/cgoflags_windows.go
+++ b/cgoflags_windows.go
@@ -1,0 +1,8 @@
+// +build windows
+
+package python
+
+// #cgo CFLAGS: -IC:/Python27/include
+// #cgo LDFLAGS: -LC:/Python27/libs -lpython27
+// #include "go-python.h"
+import "C"

--- a/cgoflags_windows.go
+++ b/cgoflags_windows.go
@@ -2,7 +2,11 @@
 
 package python
 
-// #cgo CFLAGS: -IC:/Python27/include
-// #cgo LDFLAGS: -LC:/Python27/libs -lpython27
+// #cgo 386   CFLAGS: -I C:/Python27/include
+// #cgo amd64 CFLAGS: -I C:/Python27-x64/include
+// #cgo amd64 CFLAGS: -D MS_WIN64
+// #cgo 386   LDFLAGS: -L C:/Python27/libs     -lpython27
+// #cgo amd64 LDFLAGS: -L C:/Python27-x64/libs -lpython27
+//
 // #include "go-python.h"
 import "C"

--- a/gen-cgoflags.go
+++ b/gen-cgoflags.go
@@ -1,0 +1,141 @@
+// +build ignore
+
+// simple command to generate CGOFLAGS for a given python VM
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func main() {
+	vm := flag.String("vm", "python", "path to a python VM")
+	flag.Parse()
+
+	if *vm == "" {
+		log.Fatalf("need a python VM")
+	}
+
+	cfg, err := getPythonConfig(*vm)
+	if err != nil {
+		log.Fatalf("could not infer python configuration: %v", err)
+	}
+
+	oname := "cgoflags_unix.go"
+	switch runtime.GOOS {
+	case "windows":
+		oname = "cgoflags_windows.go"
+	}
+
+	err = ioutil.WriteFile(oname, []byte(fmt.Sprintf(tmpl,
+		cfg.cflags,
+		cfg.ldflags,
+		runtime.GOOS,
+	)), 0644)
+	if err != nil {
+		log.Fatalf("could not write %q: %v", oname, err)
+	}
+}
+
+// getPythonConfig returns the needed python configuration for the given
+// python VM (python, python2, python3, pypy, etc...)
+func getPythonConfig(vm string) (pyconfig, error) {
+	code := `import sys
+import distutils.sysconfig as ds
+import json
+print(ds.get_config_vars())
+print(ds.get_python_lib())
+print(json.dumps({
+	"version": sys.version_info.major,
+	"prefix":  ds.get_config_var("prefix"),
+	"incdir":  ds.get_python_inc(),
+	"libdir":  ds.get_config_var("LIBDIR"),
+	"libdest":  ds.get_config_var("LIBDEST"),
+	"libpy":   ds.get_config_var("LIBRARY"),
+	"shlibs":  ds.get_config_var("SHLIBS"),
+	"syslibs": ds.get_config_var("SYSLIBS"),
+	"shlinks": ds.get_config_var("LINKFORSHARED"),
+	"DLLs": ds.get_config_var("DLLLIBRARY"),
+}))
+`
+
+	var cfg pyconfig
+	bin, err := exec.LookPath(vm)
+	if err != nil {
+		return cfg, errors.Wrapf(err, "could not locate python vm %q", vm)
+	}
+
+	buf := new(bytes.Buffer)
+	cmd := exec.Command(bin, "-c", code)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = buf
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		return cfg, errors.Wrap(err, "could not run python-config script")
+	}
+
+	log.Printf("distutils:\n%s", buf.String())
+
+	var raw struct {
+		Version int    `json:"version"`
+		IncDir  string `json:"incdir"`
+		LibDir  string `json:"libdir"`
+		LibPy   string `json:"libpy"`
+		ShLibs  string `json:"shlibs"`
+		SysLibs string `json:"syslibs"`
+	}
+	err = json.NewDecoder(buf).Decode(&raw)
+	if err != nil {
+		return cfg, errors.Wrapf(err, "could not decode JSON script output")
+	}
+
+	if strings.HasSuffix(raw.LibPy, ".a") {
+		raw.LibPy = raw.LibPy[:len(raw.LibPy)-len(".a")]
+	}
+	if strings.HasPrefix(raw.LibPy, "lib") {
+		raw.LibPy = raw.LibPy[len("lib"):]
+	}
+
+	cfg.version = raw.Version
+	cfg.cflags = strings.Join([]string{
+		"-I " + raw.IncDir,
+	}, " ")
+	cfg.ldflags = strings.Join([]string{
+		"-L " + raw.LibDir,
+		"-l " + raw.LibPy,
+		raw.ShLibs,
+		raw.SysLibs,
+	}, " ")
+
+	return cfg, nil
+}
+
+type pyconfig struct {
+	version int
+	cflags  string
+	ldflags string
+}
+
+const tmpl = `// Automatically generated. Do not edit.
+
+// +build %[3]s
+
+package python
+
+// #cgo %[3]s CFLAGS: %[1]s
+// #cgo %[3]s LDFLAGS: %[2]s
+//
+// #include "go-python.h"
+import "C"
+`

